### ISSUE TITLE
Assertation Failed: Unexpected exception during Attach

### ISSUE THREAD

**Issue by thedukeofcool** [CosmosOS/Cosmos#2360]
**Exception**
![](https://cdn.discordapp.com/attachments/833978908189392916/1002336351838019664/unknown.png)

**To Reproduce**
Install the latest devkit on VS2022.
Attempt to debug a C# Kernel.

**Comment by quajak** [comment]
Where is vmware installed? Can the current user launch it?

**Comment by thedukeofcool** [comment]
Yes, the current user can launch it. I've also tried running Visual Studio as admin, but that didn't work.

**Comment by quajak** [comment]
Looking at the code https://github.com/CosmosOS/Cosmos/blob/4a30847dec29c98aa0a445d69eb8067a2e9574d1/source/Cosmos.Debug.Hosts/VMware.cs#L70-L92
it is failing when trying to launch vmware. Could you try install vmware at some other location?

**Comment by thedukeofcool** [comment]
Sure, sorry I took so long.

**Comment by valentinbreiz** [comment]
Cosmos Gen3 has replaced the IL2CPU-based Gen2 codebase, and `gen3` is now the default branch of this repository. This issue was filed against Gen2, which is no longer developed, so it is being closed as part of the Gen3 cutover (#3088).

If you still hit this on Gen3, please open a fresh issue. The compiler, the build pipeline and the debugging story are all different now, so a new report will get a far more useful answer than reviving this one. Start here: https://cosmosos.github.io/

Gen2 remains on the `gen2` branch, and this issue stays readable.
